src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -55,13 +54,32 @@ private static void setupCommandTests(Application application) {
 
 		application.addTest("Visitor test DriverCommand", event -> {
 			DriverCommand command = new OperateToCommand(0, 0);
-			Visitor visitor = new DriverCommandVisitor();
-			((OperateToCommand) command).accept(visitor);
+			DriverCommandCallCounterVisitor visitor = new DriverCommandCallCounterVisitor();
+			command.accept(visitor);
+			System.out.println("Counter: " + visitor.getCounter());
 		});
 		application.addTest("Visitor test ICompoundCommand", event -> {
-			TestCompoudCommand command = new TestCompoudCommand();
-			Visitor visitor = new DriverCommandVisitor();
+			ICompoundCommand command = new ICompoundCommand() {
+				private List<DriverCommand> commands = Arrays.asList(new SetPositionCommand(0,0), new SetPositionCommand(0,0), new SetPositionCommand(0,0));
+
+				@Override
+				public Iterator<DriverCommand> iterator() {
+					return commands.iterator();
+				}
+
+				@Override
+				public void execute(Job2dDriver driver) {
+
+				}
+
+				@Override
+				public void accept(Visitor visitor) {
+					commands.forEach(command -> command.accept(visitor));
```

Comment:
This is no a correct accept implementation. You can provide the interface ICompoundCommand with default implementation of accept. 
Iteration should be a part of visit(ICompoundCommand) method in concrete visitors.